src/extensions/surveys.tsx

```diff
@@ -82,22 +83,44 @@ export const callSurveys = (posthog: PostHog, forceReload: boolean = false) => {
                     }
                 }
             }
-            if (
-                survey.type === SurveyType.Popover &&
-                document.querySelectorAll("div[class^='PostHogSurvey']").length === 0
-            ) {
+
+            if (survey.type === SurveyType.Popover) {
                 const surveyWaitPeriodInDays = survey.conditions?.seenSurveyWaitPeriodInDays
-                const lastSeenSurveyDate = localStorage.getItem(`lastSeenSurveyDate`)
-                if (surveyWaitPeriodInDays && lastSeenSurveyDate) {
-                    const today = new Date()
-                    const diff = Math.abs(today.getTime() - new Date(lastSeenSurveyDate).getTime())
-                    const diffDaysFromToday = Math.ceil(diff / (1000 * 3600 * 24))
-                    if (diffDaysFromToday < surveyWaitPeriodInDays) {
+                if (surveyWaitPeriodInDays && shouldHideSurveyInWaitPeriod(surveyWaitPeriodInDays)) {
+                    return
+                }
+                const hasEvents = survey.events && survey.events.length > 0
```

Comment:
This isn't the right place to do these checks for events imo, they need to go into the API so people using the API don't need to manually check for event triggers themselves.

This needs to go in here -> https://github.com/PostHog/posthog-js/blob/main/src/posthog-surveys.ts#L73 -> thus we can add listeners for all events for all surveys as soon as we load all surveys (I'd go as far as to say this should, in the future, be able to match on events that occurred in this session, but before we managed to load the surveys script, and also handle multiple events, both of which need a similar handling here).

We then also need some sort of listener mechanism that tells us when new surveys are active / event triggers are hit, so we can listen on those here, and accordingly show the right popup. I'm unclear how exactly this should look like, but pretty sure it's worth it to prevent any unexpected breaking for users using API surveys.